src/lightning_lite/strategies/parallel.py

```diff
@@ -0,0 +1,129 @@
+from typing import Any, Dict, Generator, List, Optional
+
+import torch
+from torch import Tensor
+from torch.nn import Module
+
+import lightning_lite as lite
+from lightning_lite.accelerators.accelerator import Accelerator
+from lightning_lite.plugins.environments.cluster_environment import ClusterEnvironment
+from lightning_lite.plugins.io.checkpoint_plugin import CheckpointIO
+from lightning_lite.plugins.precision import Precision
+from lightning_lite.strategies.strategy import Strategy
+from lightning_lite.utilities.distributed import (
+    _get_process_group_backend_from_env,
+    all_gather_ddp_if_available,
+    get_default_process_group_backend_for_device,
+    ReduceOp,
+)
+from lightning_lite.utilities.rank_zero import rank_zero_deprecation
+
+
+class ParallelStrategy(Strategy, ABC):
+    """Strategy for training with multiple processes in parallel."""
+
+    def __init__(
+        self,
+        accelerator: Optional[Accelerator] = None,
+        parallel_devices: Optional[List[torch.device]] = None,
+        cluster_environment: Optional[ClusterEnvironment] = None,
+        checkpoint_io: Optional[CheckpointIO] = None,
+        precision_plugin: Optional[Precision] = None,
+    ):
+        super().__init__(accelerator=accelerator, checkpoint_io=checkpoint_io, precision_plugin=precision_plugin)
+        self.parallel_devices = parallel_devices
+        self.cluster_environment = cluster_environment
+
+    @property
+    @abstractmethod
+    def root_device(self) -> torch.device:
+        """Return the root device."""
+
+    @property
+    def global_rank(self) -> int:
+        return self.cluster_environment.global_rank() if self.cluster_environment is not None else 0
+
+    @property
+    def local_rank(self) -> int:
+        return self.cluster_environment.local_rank() if self.cluster_environment is not None else 0
+
+    @property
+    def node_rank(self) -> int:
+        return self.cluster_environment.node_rank() if self.cluster_environment is not None else 0
+
+    @property
+    def world_size(self) -> int:
+        return self.cluster_environment.world_size() if self.cluster_environment is not None else 1
+
+    @property
+    def is_global_zero(self) -> bool:
+        return self.global_rank == 0
+
+    @property
+    def parallel_devices(self) -> Optional[List[torch.device]]:
+        return self._parallel_devices
+
+    @parallel_devices.setter
+    def parallel_devices(self, parallel_devices: Optional[List[torch.device]]) -> None:
+        self._parallel_devices = parallel_devices
+
+    @property
+    def distributed_sampler_kwargs(self) -> Dict[str, Any]:
+        distributed_sampler_kwargs = dict(
+            num_replicas=len(self.parallel_devices) if self.parallel_devices is not None else 0, rank=self.global_rank
+        )
+        return distributed_sampler_kwargs
+
+    @property
+    def torch_distributed_backend(self) -> str:
+        """Deprecated property."""
+        rank_zero_deprecation(
```

Comment:
Why not keep the deprecations in a thin subclass in PL?

Reply:
Oh, I forgot to remove this. This can safely be removed in Lite and PL will remove theirs also for 1.8.